### ISSUE TITLE
Issue with convert_raw_ngsim_to_trajdatas()

### ISSUE THREAD

**Issue by karenacai** [tawheeler/NGSIM.jl#11]
Hello, 

I have cloned all the necessary packages that were listed in the installation directions. When I try to run the command convert_raw_ngsim_to_trajdatas(), I get the following error: 

ERROR: SystemError: opening file /home/karena/.julia/v0.6/NGSIM/data/trajdata_/home/karena/.julia/v0.6/NGSIM/data/i101_trajectories-0750am-0805am.txt: No such file or directory

This error came from line 206 of trajdata.jl, which is the following: 
        open(io->write(io, MIME"text/plain"(), trajdata), outpath, "w")
where outpath is defined in line 205
        outpath = Pkg.dir("NGSIM", "data", "trajdata_"*filename)

I am relatively new to Julia, but it seems as though the outpath given by the file in the error message (/home/karena/.julia/v0.6/NGSIM/data/trajdata_/home/karena/.julia/v0.6/NGSIM/data/i101_trajectories-0750am-0805am.txt) does not exist and it must exist for the open command in Julia to work. 

Am I right to say the intention of line 206 is to output the trajdata into an output file with a specific file format that can be inputted into the AutomotiveDrivingModels.jl program? What might I be doing wrong? Thanks in advance!

**Comment by tawheeler** [comment]
Hi @karenacai, you are totally right. Please extract the data in data.zip from https://github.com/sisl/NGSIM.jl/releases/tag/v1.0.0.

**Comment by karenacai** [comment]
So I had already extracted the additional data files from that folder and placed them in the NGSIM data folder. I don't think that is the issue. The trajdata had already been formulated by the time of the error.

The issue is with writing that data to an output file, which I think is what you're doing with the open(io->write...) command. The output path has the directory path to the NGSIM folder, followed by "/traj_data", followed by the filename, which is the file that does not exist?

**Comment by saArbabi** [comment]
Hi @karenacai, did you solve the problem? I am having a similar issue with error: 

SystemError: opening file trajdata_C:\Users\sa00443\.julia\v0.6\NGSIM\data\i101_trajectories-0750am-0805am.txt: Invalid argument

If you managed to solve it, could you let me know how?

Thanks


PS: I also  have already extracted the additional data files from [Data ](https://github.com/sisl/NGSIM.jl/releases )folder and placed them in the NGSIM data folder.

**Comment by tawheeler** [comment]
@karenacai Looks like `filename` is `home/karena/.julia/v0.6/NGSIM/data/i101_trajectories-0750am-0805am.txt`, which is already a complete directory path and doesn't need "/home/karena/.julia/v0.6/NGSIM/data/trajdata_" prepended to it.
Maybe use filename instead of outpath in like 206?

@saArbabi Your filepath also looks bad. I'd expect "C:\Users\sa00443.julia\v0.6\NGSIM\data\i101_trajectories-0750am-0805am.txt". Any idea why it is constructing it that way?

**Comment by saArbabi** [comment]
@tawheeler Thanks for you reply. I am not sure what is causing the issue (I am new to Julia). Could it be the operating system? I am on windows machine and I wonder if the issue doesn't arise say on Linux. 

I will play with it and if the issue is solved I will report back.

**Comment by tawheeler** [comment]
Looks like `NGSIM_TRAJDATA_PATHS` contains full paths rather than just file names.

I pushed an update to get the package more up to date with julia 1.0. I verified that I could convert at least the first file. 
It is quite possible that more of the package (and maybe automotive driving models in general) needs to be updated to Julia 1.0.

**Comment by saArbabi** [comment]
Yes @tawheeler. Assigning the actual Trajdata_ file name I can convert one data file at a time.

**Comment by fei960922** [comment]
Hi author~

I follow the instruction and copy the data into right place but error occured:
'''
julia> convert_raw_ngsim_to_trajdatas()
converting i101_trajectories-0750am-0805am.txt
loading from file: ┌ Warning: readtable is deprecated, use CSV.read from the CSV package instead
│   caller = ip:0x0
└ @ Core :-1
ERROR: MethodError: no method matching rename!(::DataFrames.Index, ::Symbol, ::Symbol)
Closest candidates are:
  rename!(::DataFrames.Index, ::Any) at /home/fei960922/.julia/packages/DataFrames/VrZOl/src/other/index.jl:49
  rename!(::DataFrames.AbstractDataFrame, ::Any...) at /home/fei960922/.julia/packages/DataFrames/VrZOl/src/abstractdataframe/abstractdataframe.jl:129
  rename!(::DataFrames.Index, ::Pair{Symbol,Symbol}...) at /home/fei960922/.julia/packages/DataFrames/VrZOl/src/other/index.jl:60
Stacktrace:
 [1] rename!(::DataFrames.DataFrame, ::Symbol, ::Symbol) at /home/fei960922/.julia/packages/DataFrames/VrZOl/src/abstractdataframe/abstractdataframe.jl:129
 [2] NGSIMTrajdata(::String) at /home/fei960922/.julia/packages/NGSIM/OPF1x/src/ngsim_trajdata.jl:37
 [3] #load_ngsim_trajdata#35(::Bool, ::Function, ::String) at /home/fei960922/.julia/packages/NGSIM/OPF1x/src/trajdata.jl:146
 [4] load_ngsim_trajdata at /home/fei960922/.julia/packages/NGSIM/OPF1x/src/trajdata.jl:145 [inlined]
 [5] convert_raw_ngsim_to_trajdatas() at /home/fei960922/.julia/packages/NGSIM/OPF1x/src/trajdata.jl:200
 [6] top-level scope at none:0
'''

Do you know what happened? Thanks.


Sorry. Found solution in another issue. It is due to new julia version. See: https://github.com/sisl/ngsim_env/issues/13#issue-441241017

**Comment by tawheeler** [comment]
PRs to improve the package would be welcomed!

**Comment by DarrenRuan** [comment]
HI my Julia Version is 1.4, macOS
I met this error when I did using NGSIM, and I have no idea about what happened here:

```
julia> using NGSIM
[ Info: Precompiling NGSIM [195157bc-cedd-5875-9ea7-7cfb75f30b3c]
┌ Warning: This package is deprecated: use AutomotiveSimulator.jl instead.
└ @ AutomotiveDrivingModels ~/.julia/packages/AutomotiveDrivingModels/MMTAw/src/AutomotiveDrivingModels.jl:3
ERROR: LoadError: LoadError: MethodError: no method matching read(::IOStream, ::MIME{Symbol("text/plain")}, ::Type{AutomotiveDrivingModels.Roadway})
Closest candidates are:
  read(::AbstractString, ::Any...) at io.jl:379
  read(::IOStream) at iostream.jl:453
  read(::IOStream, ::Type{UInt8}) at iostream.jl:348
  ...
Stacktrace:
```

What should I do to run convert_raw_ngsim_to_trajdatas()? I could not even using NGSIM. Is this a issue of a new version of Julia?

PS: I have also already extracted the additional data files from Data folder and placed them in the NGSIM data folder.
